x-pack/plugins/case/server/connectors/servicenow/sir_formatter.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get } from 'lodash/fp';
+import { ConnectorServiceNowSIRTypeFields } from '../../../common/api';
+import { ExternalServiceFormatter } from '../types';
+
+interface ExternalServiceParams {
+  dest_ip: string | null;
+  source_ip: string | null;
+  category: string | null;
+  subcategory: string | null;
+  malware_hash: string | null;
+  malware_url: string | null;
+  priority: string | null;
+}
+
+type AlertFieldMappingAndValues = Record<
+  string,
+  { alertPath: string; values: Set<string>; sirFieldKey: string }
+>;
+
+const format: ExternalServiceFormatter<ExternalServiceParams>['format'] = async (
+  theCase,
+  alerts
+) => {
+  const {
+    destIp = null,
+    sourceIp = null,
+    category = null,
+    subcategory = null,
+    malwareHash = null,
+    malwareUrl = null,
+    priority = null,
+  } = (theCase.connector.fields as ConnectorServiceNowSIRTypeFields['fields']) ?? {};
+  const alertFieldsVariables = [destIp, sourceIp, malwareHash, malwareUrl];
+
+  // Set is used to avoid duplicates
+  const alertFieldMappingAndValues: AlertFieldMappingAndValues = {
+    destIp: { alertPath: 'destination.ip', values: new Set(), sirFieldKey: 'dest_ip' },
+    sourceIp: { alertPath: 'source.ip', values: new Set(), sirFieldKey: 'source_ip' },
+    malwareHash: { alertPath: 'file.hash.sha256', values: new Set(), sirFieldKey: 'malware_hash' },
+    malwareUrl: { alertPath: 'url.full', values: new Set(), sirFieldKey: 'malware_url' },
+  };
+
+  if (alertFieldsVariables.some((field) => field != null)) {
+    alerts.forEach((alert) => {
```

Comment:
can we talk about this logic? I feel we can simplify it to not go through all the alerts for every `alertFieldsVariables` but do it the other way around. I feel like we will have more alerts than fields.

Reply:
We are not going through the alerts for every field. We iterate the alerts and we check if the alert has the field.

Reply:
I miss read it, all good